lite/proxy/block.go

```diff
@@ -11,11 +11,17 @@ import (
 )
 
 func ValidateBlockMeta(meta *types.BlockMeta, check lite.Commit) error {
+	if meta == nil {
+		return errors.New("expecting a non-nil BlockMeta")
```

Comment:
Would it make sense to consolidate error handling in lite? There is [lite/errors](https://github.com/tendermint/tendermint/tree/master/lite/errors) which is used in this file further down as `certerr` and there is [lite/proxy/errors.go](https://github.com/tendermint/tendermint/blob/master/lite/proxy/errors.go).

A more rigid approach to errors would enable control flow for callers.

Reply:
@xla 
>Consolidate error handling in lite

Thank you for raising that, I'll do that.

>A more rigid approach to errors would enable control flow for callers
What does this mean?

Reply:
Constructing meaningful errors that the caller can vary on will help to make decisions upstream and will lead to a safer way of interacting with the code, e.g.:

``` go
// meta := ...
// commit := ...
err := proxy.ValidateBlockMeta(meta, commit)
switch errors.Cause(err) {
case proxy.ErrNiBlock:
    // reconsile
case proxy.ErrNilHeader:
    // reconsile
case proxy.ErrHeadersNotMatch:
    // reconsile
default:
    // If we really have to propagate: return err
}
```

If the errors should originate from `proxy`, `lite` or `lite/errors` is something we can play with. Usually the more central the less overhead per set of errors to maintain. A mix is also possible: `proxyErr.Err...`, `liteErr.Err...` as long as we consistently wrap and unpack.

Reply:
It has sentinesl, but unexported and uses a lot of support boilerplate to test for the errors. We should simplify that package and have proper sentinels.

Reply:
@xla as per discussions below, that's more comprehensive work that'll be tackled in another issue https://github.com/tendermint/tendermint/issues/1360.